server/modlog.ts

```diff
@@ -36,284 +32,255 @@ const PUNISHMENTS = [
 	'NOTE', 'MAFIAHOSTBAN', 'MAFIAUNHOSTBAN', 'GIVEAWAYBAN', 'GIVEAWAYUNBAN',
 	'TOUR BAN', 'TOUR UNBAN', 'UNNAMELOCK',
 ];
-const PUNISHMENTS_REGEX_STRING = `\\b(${PUNISHMENTS.join('|')}):.*`;
-
-const execFile = util.promisify(child_process.execFile);
 
 export type ModlogID = RoomID | 'global';
 
 interface ModlogResults {
-	results: string[];
+	results: ModlogEntry[];
 	duration?: number;
 }
 
 interface ModlogQuery {
-	rooms: ModlogID[];
-	search: string;
-	isExact: boolean;
-	maxLines: number;
-	onlyPunishments: boolean | string;
+	statement: Database.Statement;
+	args: any[];
 }
 
-class SortedLimitedLengthList {
-	maxSize: number;
-	list: string[];
+export interface ModlogSearch {
+	note?: {searches: string[], isExact?: boolean};
+	user?: string;
+	ip?: string;
+	action?: string;
+	actionTaker?: string;
+}
 
-	constructor(maxSize: number) {
-		this.maxSize = maxSize;
-		this.list = [];
-	}
+export interface ModlogEntry {
+	action: string;
+	roomID?: string;
+	visualRoomID?: string;
+	userid?: ID;
+	autoconfirmedID?: ID;
+	alts?: ID[];
+	ip?: string;
+	isGlobal?: boolean;
+	loggedBy?: ID;
+	note?: string;
+	/** Milliseconds since the epoch */
+	time?: number;
+}
 
-	getListClone() {
-		return this.list.slice();
-	}
+export class Modlog {
+	readonly database: Database.Database;
+	readonly usePM: boolean;
+	readonly modlogInsertionQuery: Database.Statement;
+	readonly altsInsertionQuery: Database.Statement;
+	readonly renameQuery: Database.Statement;
+	readonly globalPunishmentsSearchQuery: Database.Statement;
+	readonly insertionTransaction: Database.Transaction;
```

Comment:
Should be able to type the parameters these queries expect by providing an argument to `Statement`/`Transaction`